src/firebase.tsx

```diff
@@ -125,6 +134,15 @@ export function SubscribeToTiles(onNewTiles: (newTiles: TileDict) => void ): Voi
   });
 }
 
+// Adds tile with specified properties
+export async function AddTile(tileName: string, tileInfo: TileInfo) {
+  try {
+    await firestore.collection("tiles").doc(tileName).set(tileInfo);
+  } catch {
+    console.error("Update failed");
```

Comment:
This probably wants to re-raise the error if you can't create a new tile (which would likely be caused by re-using a name).